crates/polars-plan/src/logical_plan/builder.rs

```diff
@@ -683,19 +683,42 @@ impl LogicalPlanBuilder {
             into
         );
 
+        // Initialize schema from keys
         let mut schema = try_delayed!(
             expressions_to_schema(&keys, current_schema, Context::Default),
             &self.0,
             into
         );
-        let other = try_delayed!(
+
+        // Add dynamic groupby index column(s)
+        #[cfg(feature = "dynamic_group_by")]
+        {
+            if let Some(options) = rolling_options.as_ref() {
+                let name = &options.index_column;
+                let dtype = try_delayed!(current_schema.try_get(name), self.0, into);
+                schema.with_column(name.clone(), dtype.clone());
```

Comment:
Not entirely sure, but I believe we must insert those at the start of the schema?

Reply:
First are the "keys" (specified in the `by` parameter), then are the `_lower_bound`/`_upper_bound` columns if specified, then the index column, and then the aggregation columns.

This is the way it's currently implemented, I am just making the schema fit. I have to admit I was scratching my head a little bit about this ordering myself...

Reply:
What would make sense to me is:

* Index column (e.g. `time`)
* Bounds if `include_bounds=True`, named after the index column (e.g. `time_lower_bound`, `time_upper_bound`)
* Additional groups if `by` is specified (e.g. `group1`)
* Aggregations (e.g. `agg1`).

So if all are present you get `[time, time_lower_bound, time_upper_bound, group1, group2, ..., agg1, agg2, ...]`

If that sounds more sensible than the status quo, now is the time to change it 😄

Reply:
I mean more. Do we return the same ordering as the materialization? That's what is important. I didn't want to break. :smile:

Reply:
We do! There are tests for that. So this can be merged.